how-to/workspace-platform-starter/client/src/framework/launch.ts

```diff
@@ -372,12 +382,99 @@ async function launchWindow(windowApp: PlatformApp): Promise<PlatformAppIdentifi
 
 	if (!windowExists) {
 		try {
+			const appLaunchPreference = objectClone(windowApp.launchPreference);
+			const appLaunchPreferenceOptions = objectClone(
+				windowApp.launchPreference?.options
+			) as WindowLaunchOptions;
+
+			// validate passed launch preferences
+			if (
+				appLaunchPreference?.options?.type === "window" &&
+				Array.isArray(appLaunchPreferenceOptions?.updatable) &&
+				appLaunchPreferenceOptions.updatable.length > 0
+			) {
+				console.log(appLaunchPreference.options.updatable);
+				for (const option of appLaunchPreferenceOptions.updatable) {
+					if (option.name === "BOUNDS") {
+						appLaunchPreference.bounds = launchPreference?.bounds;
+					}
+					if (option.name === "CENTERED") {
+						appLaunchPreference.defaultCentered = launchPreference?.defaultCentered;
+					}
+					if (
+						option.name === "CUSTOM_DATA" &&
+						launchPreference?.options?.type === "window" &&
+						!isEmpty(launchPreference.options?.window?.customData)
+					) {
+						if (isEmpty(appLaunchPreferenceOptions.window)) {
+							appLaunchPreferenceOptions.window = {};
+						}
+						appLaunchPreferenceOptions.window.customData = {
+							...appLaunchPreferenceOptions.window?.customData,
+							...launchPreference.options?.window?.customData
+						};
+					}
+					if (
+						option.name === "INTEROP" &&
+						launchPreference?.options?.type === "window" &&
+						!isEmpty(launchPreference.options?.window?.interop)
+					) {
+						if (isEmpty(appLaunchPreferenceOptions.window)) {
+							appLaunchPreferenceOptions.window = {};
+						}
+						appLaunchPreferenceOptions.window.interop = launchPreference?.options.window?.interop;
+					}
+					if (
+						option.name === "URL" &&
+						launchPreference?.options?.type === "window" &&
+						isStringValue(launchPreference.options?.window?.url)
+					) {
+						const suggestedUrl: string = launchPreference.options?.window?.url as string;
```

Comment:
If you assign the url to a var, then validate and use it, there is no need to cast i.e.
```
const suggestedUrl = launchPreference.options?.window?.url;
if (isStringValue(suggestedUrl)) {
...
}
```

Reply:
tidied up assigning the urls.